inference-engine/thirdparty/clDNN/kernel_selector/core/kernel_selector.cpp

```diff
@@ -98,8 +100,12 @@ KernelsData kernel_selector_base::GetNaiveBestKernel(const Params& params,
                 }
 #endif
             }
-        } catch (std::runtime_error&) {
+        } catch (std::runtime_error& ex) {
             // we have to handle it in order to avoid exception in KernelSelector as much we can
+            GPU_DEBUG_GET_INSTANCE(debug_config);
+            GPU_DEBUG_IF(debug_config->verbose >= MIN_VERBOSE_LEVEL) {
+                GPU_DEBUG_COUT << ex.what() << std::endl;
```

Comment:
I guess that this execption will happen frequently because it is not error, but just a selection failure for a specific kernel. However if you want to enable to check each kernels failing status, I think you need  to add the kernel name here too. Otherwise the log will just have execeptions..

Reply:
To be more informative, "primitive id" and the finally selected kernel is also needed to be given with this verbose log.

Reply:
Fixed it. thanks